packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -248,29 +247,29 @@ export class SharedTreeBranch<
 
 	/**
 	 * Apply a change to this branch.
-	 * @param change - the change to apply
+	 * @param taggedChange - the change to apply
 	 * @param revision - the revision of the new head commit of the branch that contains `change`
 	 * @param changeKind - the kind of change to apply
 	 * @returns the change that was applied and the new head commit of the branch
 	 */
 	public apply(
-		change: TChange,
-		revision: RevisionTag,
+		taggedChange: TaggedChange<TChange>,
+		revision: RevisionTag, // This can be removed once we make revision required on taggedChange?
 		changeKind: CommitKind = CommitKind.Default,
 	): [change: TChange, newCommit: GraphCommit<TChange>] {
 		this.assertNotDisposed();
 
-		// TODO: This should not be necessary when receiving changes from other clients.
-		const changeWithRevision = this.changeFamily.rebaser.changeRevision(change, revision);
+		const revisionTag = taggedChange.revision;
+		assert(revisionTag !== undefined, "Revision tag must be provided");
```

Comment:
One difference between the previous `changeWithRevision` and the new one is that the `revisions` property in the change has multiple entries with the same value. For example, this is what it looks like now:
<img width="341" alt="image" src="https://github.com/user-attachments/assets/e078f793-6d79-4498-8cca-c4558521c50d">

Previously, there was only one entry. It doesn't seem to matter but I am not very sure.

Reply:
This may not break anything right now, but it's definitely unexpected so it would be good to at least understand where it comes from.

Reply:
It's because when `buildChanges` is now called with a list of changes [here](https://github.com/microsoft/FluidFramework/blob/1d2fe127cd16eed0eeae343309e68ff63d45877d/packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts#L2612), each of which has a revision associated with it.
When the modular change sets generated in `buildChanges` are composed, it composes these revisions into a list. I looked at that code and it's possible to remove duplicates during compose. But I am not sure if that's the right place to do it or should it be taken care of before compose is called.

Reply:
Rmeoved the duplicates in `getRevInfoFromTaggedChanges` in `modularChangeFamily.ts`